src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,132 @@
+            _localizer[key, arguments];
+
+        /// <summary>
+        /// Creates a new <see cref="IStringLocalizer"/> for a specific <see cref="CultureInfo"/>.
+        /// </summary>
+        /// <param name="culture">The <see cref="CultureInfo"/> to use.</param>
+        /// <returns>A culture-specific <see cref="IStringLocalizer"/>.</returns>
+        IStringLocalizer IStringLocalizer.WithCulture([NotNull] CultureInfo culture) => WithCulture(culture);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key) => _localizer.GetString(key);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key, params object[] arguments) =>
+            _localizer.GetString(key, arguments);
+
+        /// <inheritdoc />
+        public IEnumerable<LocalizedString> GetAllStrings(bool includeAncestorCultures) =>
+            _localizer.GetAllStrings(includeAncestorCultures);
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key) => ToHtmlString(_localizer.GetString(key));
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key, params object[] arguments) =>
+            ToHtmlString(_localizer.GetString(key, EncodeArguments(arguments)));
+
+        /// <summary>
+        /// Creates a new <see cref="IHtmlLocalizer"/> for a specific <see cref="CultureInfo"/>.
+        /// </summary>
+        /// <param name="culture">The <see cref="CultureInfo"/> to use.</param>
+        /// <returns>A culture-specific <see cref="IHtmlLocalizer"/>.</returns>
+        public IHtmlLocalizer WithCulture([NotNull]CultureInfo culture) =>
+            new HtmlLocalizer(_localizer.WithCulture(culture), _encoder);
+
+        /// <summary>
+        /// Creates a new <see cref="LocalizedHtmlString"/> for a <see cref="LocalizedString"/>.
+        /// </summary>
+        /// <param name="result">The <see cref="LocalizedString"/>.</param>
+        protected LocalizedHtmlString ToHtmlString(LocalizedString result) =>
+            new LocalizedHtmlString(result.Name, result.Value, result.ResourceNotFound);
+
+        /// <summary>
+        /// Encodes the arguments based on the object type.
+        /// </summary>
+        /// <param name="arguments">The array of objects to encode.</param>
+        /// <returns>The encoded object array.</returns>
+        protected object[] EncodeArguments(object[] arguments)
+        {
+            var encodedArguments = new object[arguments.Length];
+            for (var index = 0; index != arguments.Length; ++index)
+            {
+                var argument = arguments[index];
+                if (argument is HtmlString)
+                {
+                    encodedArguments[index] = argument;
+                }
+                else
+                {
+                    encodedArguments[index] = _encoder.HtmlEncode(argument.ToString());
```

Comment:
This will `throw` if `argument == null`. Even in non-error situations, it's probably not what users want because it means type-specific formatting such as `{0d}` won't work.

With this and the "must-fix" issue above, I lean toward returning simple `string`s and leaving everything about `HtmlString`s out of this PR. If we find scenarios where values get double-encoded (i.e. arguments are already `HtmlString`s), we can correct those code paths. Leave HTML encoding to Razor page and so on.